src/table2qb/configuration/column.clj

```diff
@@ -54,60 +56,67 @@
-
-(defn parse-column [{:keys [title name] :as row}]
-  (validate-title title)
-  (validate-column-name name)
-  (let [{vt :value_transformation :as normalised} (util/map-values row util/blank->nil)
-        optional-keys [:property_template :value_template :datatype :value_transformation]]
-    (merge
-      {:title                title
-       :name                 name
-       :type                 (get-column-type (:component_attachment normalised))}
-      (util/filter-vals some? (select-keys normalised optional-keys))
-      (when vt
-        {:value_transformation (resolve-value-transformation vt)}))))
+(def ^:private attachment->type {"qb:dimension" :dimension
+                                 "qb:measure" :measure
+                                 "qb:attribute" :attribute})
+
+(defn validate-column-type [row column value]
+  (if (string/blank? value)
+    :value
+    (let [type (get attachment->type (string/trim value) ::missing)]
+      (if (= ::missing type)
+        (csv/throw-cell-validation-error row column "Value must be blank or one of qb:dimension, qb:measure or qb:attribute" {})
+        type))))
+
+(defn validate-name [row column value]
+  (if (string/includes? value "-")
+    (csv/throw-cell-validation-error row column "csvw:name cannot contain hyphens (use underscores instead)" {})
+    value))
+
+(defn validate-csvw-datatype [row column value]
+  (if (contains? datatype/type-names value)
+    value
+    (csv/throw-cell-validation-error row column (format "Invalid XML datatype name: '%s'" value) {})))
+
+(defn uri-template
+  "Validates the given cell value contains a valid URI template."
+  [row column value]
+  (if-let [_template (uri-template/try-parse-template value)]
+    value
+    (csv/throw-cell-validation-error row column (format "Could not parse %s as a URI template" value) {})))
+
+(def csv-columns [{:title    "title"
+                   :key      :title
+                   :validate [csv/validate-not-blank]
+                   :required true}
+                  {:title "name"
+                   :key :name
+                   :required true
+                   :validate [csv/validate-not-blank validate-name]}
+                  {:title     "component_attachment"
+                   :key       :type
+                   :transform validate-column-type}
+                  {:title "property_template"
+                   :key :property_template
+                   :transform (csv/optional uri-template)}
+                  {:title "value_template"
+                   :key :value_template
+                   :transform (csv/optional uri-template)}
+                  {:title "datatype"
+                   :key :datatype
+                   :transform (csv/optional validate-csvw-datatype)}
+                  {:title "value_transformation"
+                   :key :value_transformation
+                   :transform (csv/optional (csv/validate-mapping column-transformers))}])
+
+(defn normalise-column-record [row]
```

Comment:
I'm a little wary of using the term "normalise". This makes me think of structural changes - e.g. un-pivoting to 3rd Normal Form.

Is this operation something more like `remove-optional-columns`?